src/managers/GuildMemberManager.js

```diff
@@ -209,23 +277,21 @@ class GuildMemberManager extends BaseManager {
    *   .then(user => console.log(`Banned ${user.username || user.id || user} from ${guild.name}`))
    *   .catch(console.error);
    */
-  ban(user, options = { days: 0 }) {
+  async ban(user, options = { days: 0 }) {
     if (typeof options !== 'object') return Promise.reject(new TypeError('INVALID_TYPE', 'options', 'object', true));
     if (options.days) options.delete_message_days = options.days;
     const id = this.client.users.resolveID(user);
     if (!id) return Promise.reject(new Error('BAN_RESOLVE_ID', true));
-    return this.client.api
-      .guilds(this.guild.id)
-      .bans[id].put({ data: options })
-      .then(() => {
-        if (user instanceof GuildMember) return user;
-        const _user = this.client.users.resolve(id);
-        if (_user) {
-          const member = this.resolve(_user);
-          return member || _user;
-        }
-        return id;
-      });
+
+    await this.client.api.guilds(this.guild.id).bans[id].put({ data: options });
+
+    if (user instanceof GuildMember) return user;
+    const _user = this.client.users.resolve(id);
+    if (_user) {
+      const member = this.resolve(_user);
+      return member || _user;
```

Comment:
```suggestion
      return this.resolve(_user) || _user;
```

Reply:
Using `??` would be better, though:
```suggestion
      return this.resolve(_user) ?? _user;
```

Reply:
Fixed in #bbfd002